### ISSUE TITLE
The Darkness II - black textures

### ISSUE THREAD

**Issue by zemfil** [iXit/Mesa-3D#264]
The Darkness II has black textures in various places. Wine has the same problem.

![darkness_black_carpet](https://cloud.githubusercontent.com/assets/23645689/21829343/b5c7e9ac-d795-11e6-8c6b-7a4fefa8b537.jpg)

The carpet in the shot above should be red for example.

[apitrace](https://mega.nz/#!bNo31C6Y!u3V0m82KX1YZTUPvl_XeJd1llwq169j0X-DGIp7-948)
[apitrace 2](https://mega.nz/#!jUhBjDyT!7bk_LsFc6EAgPITfziwn0iMeWikih9UclHnXqoD6w50) (textures not illuminated by a light source seem to be rendered in black)
[apitrace windows](https://mega.nz/#!yI4UDBTR!ZhUpc0PZJAfvyyOQ1edLGK5FDw8qSSWdZiEEYYhQR7U)

On Windows the ground textures become first visible for example on frame 680 call 1054331 (Trace 2), but still the trace replay is somewhat broken on Windows. Maybe because I took the trace from wine?

Added Windows trace.

**Comment by axeldavy** [comment]
Could you try play with override_vendorid  (it's an env var) ?

Nvidia:
override_vendorid=4318
Intel
override_vendorid=32902

**Comment by axeldavy** [comment]
Is replaying the windows trace on windows giving the correct output ?

**Comment by axeldavy** [comment]
OK, I looked at the apitrace results, and I think the problem is 0*inf = 0 dx9 behaviour.

There is discussions for an gallium extension with the behaviour and it was added to nouveau, but not yet radeonsi. Currently there is no way for either nine or wine to implement the behaviour (besides checking for 0 for any mul instruction).

Not generating infs (the intel windows way) seems to work on the trace. I'll push a patch shortly.
